tsig.go

```diff
@@ -89,6 +97,48 @@ type timerWireFmt struct {
 	Fudge      uint16
 }
 
+type tsigHMACProvider string
```

Comment:
I think move this and it's associated methods so it comes right after `type TsigProvider interface { ... }`.

Reply:
I did have it there when I first wrote it and then thought it maybe should be closer to the code that used it. I was right first time 😄